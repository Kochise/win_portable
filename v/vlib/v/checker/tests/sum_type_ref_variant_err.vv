struct Abc {
    val string
}
struct Xyz {
    foo string
}
type Alphabet1 = Abc | string | &Xyz
type Alphabet2 = Abc | &Xyz | string
type Alphabet3 = &Xyz | Abc | string