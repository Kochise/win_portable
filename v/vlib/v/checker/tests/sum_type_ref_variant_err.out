vlib/v/checker/tests/sum_type_ref_variant_err.vv:7:33: error: sum type cannot hold a reference type
    5 |     foo string
    6 | }
    7 | type Alphabet1 = Abc | string | &Xyz
      |                                 ~~~~
    8 | type Alphabet2 = Abc | &Xyz | string
    9 | type Alphabet3 = &Xyz | Abc | string
vlib/v/checker/tests/sum_type_ref_variant_err.vv:8:24: error: sum type cannot hold a reference type
    6 | }
    7 | type Alphabet1 = Abc | string | &Xyz
    8 | type Alphabet2 = Abc | &Xyz | string
      |                        ~~~~
    9 | type Alphabet3 = &Xyz | Abc | string
vlib/v/checker/tests/sum_type_ref_variant_err.vv:9:18: error: sum type cannot hold a reference type
    7 | type Alphabet1 = Abc | string | &Xyz
    8 | type Alphabet2 = Abc | &Xyz | string
    9 | type Alphabet3 = &Xyz | Abc | string
      |                  ~~~~